pkg/types/contract_reader.go

```diff
@@ -118,3 +118,47 @@ func (bc BoundContract) ReadIdentifier(readName string) string {
 func (bc BoundContract) String() string {
 	return bc.Address + "-" + bc.Name
 }
+
+type UnimplementedContractReader struct{}
+
+var _ ContractReader = UnimplementedContractReader{}
```

Comment:
The strict version of this includes an unexported method so that impls _must_ embed `UnimplementedContractReader` to satisfy the interface. Should we add that now?

Reply:
done